chore: rename LLL to IR (#2724)
vyperlang/vyper: vyper/ir/README.md

> # Introduction to IR

Author: ```suggestion
# Introduction to Vyper's IR
```

> IR is Vyper's current intermediate representation. It is used as "a high level assembly". While IR could theoretically compile to any backend which shares enough intrinsics with the EVM, this document assumes the reader has a good understanding of the EVM's execution model.

Reviewer: "IR is Vyper's current intermediate representation"

This sentence is weird to read

> IR is Vyper's current intermediate representation. It is used as "a high level assembly". While IR could theoretically compile to any backend which shares enough intrinsics with the EVM, this document assumes the reader has a good understanding of the EVM's execution model.

Author: ```suggestion
VyperIR is Vyper's current intermediate representation. It is used as "a high level assembly". While VyperIR could theoretically compile to any backend which shares enough intrinsics with the EVM, this document assumes the reader has a good understanding of the EVM's execution model.
```

> # Structure of IR

Author: ```suggestion
# Structure of VyperIR
```

> The grammar of IR is `(s_expr)` where `s_expr` is one of

Author: ```suggestion
The grammar of VyperIR is `(s_expr)` where `s_expr` is one of
```

  "repeat" s_expr s_expr s_expr s_expr s_expr
```

An IR expression has a "valency" of 1 or 0. Valency of 1 means that it returns a stack item, valency of 0 means that it does not.

> Vyper's current implementation of IR can be referenced, mainly in [vyper/codegen/ir\_node.py](../codegen/ir_node.py) (which describes the internal API for constructing IR) and in [vyper/ir/compile\_ir.py](../ir/compile_ir.py) (which describes how IR is compiled to assembly and then to bytecode). Vyper's IR output can be inspected by compiling a vyper contract with `vyper -f ir <contract.vy>`. Vyper also comes with a tool, `vyper-ir` which can be used to compile IR directly to assembly or EVM bytecode.

Reviewer 2: "Vyper's IR output can be inspected by compiling a ~~vyper~~ Vyper contract with `vyper -f ir <contract.vy>`."

> Vyper's current implementation of IR can be referenced, mainly in [vyper/codegen/ir\_node.py](../codegen/ir_node.py) (which describes the internal API for constructing IR) and in [vyper/ir/compile\_ir.py](../ir/compile_ir.py) (which describes how IR is compiled to assembly and then to bytecode). Vyper's IR output can be inspected by compiling a vyper contract with `vyper -f ir <contract.vy>`. Vyper also comes with a tool, `vyper-ir` which can be used to compile IR directly to assembly or EVM bytecode.

Author: ```suggestion
Vyper's current implementation of IR can be referenced, mainly in [vyper/codegen/ir\_node.py](../codegen/ir_node.py) (which describes the internal API for constructing IR) and in [vyper/ir/compile\_ir.py](../ir/compile_ir.py) (which describes how IR is compiled to assembly and then to bytecode). Vyper's IR output can be inspected by compiling a Vyper contract with `vyper -f ir <contract.vy>`. Vyper also comes with a tool, `vyper-ir` which can be used to compile IR (in Lisp syntax) directly to assembly or EVM bytecode.
```

> In the following examples, `_sym_<label>` is a location in code which will be resolved as the last step during conversion to opcodes. If it occurs before a `JUMPDEST`, is merely a marker in the code (and gets omitted in the bytecode). If it occurs anywhere else, it translates to `PUSH2 <location of jumpdest>`.

Reviewer 2: "If it occurs before a `JUMPDEST`, **it** is merely a marker in the code (and gets omitted in the bytecode)."

> In the following examples, `_sym_<label>` is a location in code which will be resolved as the last step during conversion to opcodes. If it occurs before a `JUMPDEST`, is merely a marker in the code (and gets omitted in the bytecode). If it occurs anywhere else, it translates to `PUSH2 <location of jumpdest>`.

Author: ```suggestion
In the following examples, `_sym_<label>` is a location in code which will be resolved as the last step during conversion to opcodes. If it occurs before a `JUMPDEST`, it is merely a marker in the code (and gets omitted in the bytecode). If it occurs anywhere else, it translates to `PUSH2 <location of jumpdest>`.
```

### GOTO

> jump to a label. Vyper’s dialect of IR has two forms, simple goto and goto with args. The latter form is useful for implementing subroutines.

Author: ```suggestion
jump to a label. VyperIR has two forms of goto, simple goto and goto with args. The latter form is useful for implementing subroutines.
```
